src/dsl/verifier.ts

```diff
@@ -158,6 +165,65 @@ export class Verifier {
     }
   }
 
+  private createRequestTracer(): express.RequestHandler {
+    return (req, _, next) => {
+      logger.trace(
+        "incoming request",
+        pickBy(
+          {
+            body: req.body,
+            headers: req.headers,
+            method: req.method,
+            path: req.path,
+          },
+          identity
+        )
+      )
+      next()
+    }
+  }
+
+  private createResponseTracer(): express.RequestHandler {
+    return (_, res, next) => {
+      const [oldWrite, oldEnd] = [res.write, res.end]
+      const chunks: Buffer[] = []
+
+      res.write = (chunk: any) => {
+        chunks.push(Buffer.from(chunk))
+        return oldWrite.apply(res, [chunk])
+      }
+
+      res.end = (chunk: any) => {
+        if (chunk) {
+          chunks.push(Buffer.from(chunk))
+        }
+        const body = Buffer.concat(chunks).toString("utf8")
+        logger.trace(
+          "outgoing response",
+          pickBy(
+            {
+              body,
+              headers: reduce(
+                res.getHeaders(),
+                (acc: any, val, index) => {
+                  acc[index] = val
+                  return acc
+                },
+                {}
+              ),
+              status: res.statusCode,
+            },
+            identity
+          )
+        )
+        oldEnd.apply(res, [chunk])
+      }
+      if (next) {
```

Comment:
This is probably fine, but would it be safer for this to check if `next` were a function?

Reply:
yep, I'll fix up all of the types before merge